app/javascript/mastodon/selectors/filters.ts

```diff
@@ -0,0 +1,58 @@
+import { createSelector } from '@reduxjs/toolkit';
+
+import type { RootState } from 'mastodon/store';
+import { toServerSideType } from 'mastodon/utils/filters';
+
+// TODO: move to `app/javascript/mastodon/models` and use more globally
+type Filter = Immutable.Record<{
```

Comment:
This is unfortunately incorrect, it's not an `Immutable.Record` but an `Immutable.map<string, unknown>`, which does not allow good typing.
We should convert this to records (or plain JS objects), but until then it is probably best to not try to type this code :(

Reply:
ok, it does work because the interfaces we use are common between `Map` and `Record`, but you're right. Amended the code to still type it correctly, but it makes pretty extensive use of `as`. At least it's self-contained.

Reply:
Correct, `Record` supports for example `record.attribute`, but `Map` only supports `record.get('attribute')`